### PR TITLE
Add description page of Dynamo IPEX backend

### PR DESCRIPTION
**Pull request by jingxu10**
As title.
Add description page of Dynamo IPEX backend

## Checklist

- [ ] The issue that is being fixed is referred in the description (see above "Fixes #ISSUE_NUMBER")
- [ ] Only one issue is addressed in this pull request
- [ ] Labels from the issue that this PR is fixing are added to this pull request
- [ ] No unnecessary issues are included into this pull request.

cc @williamwen42 @msaroufim

### CODE REVIEW

**Review by jgong5 — APPROVED** [review]

**Comment by jingxu10** [comment]
The error shows
```
Run pytorch/test-infra/.github/actions/pull-docker-image@main
Run retry () { "$@"  || (sleep 1 && "$@") || (sleep 2 && "$@") }
Error response from daemon: manifest for 30[8](https://github.com/pytorch/tutorials/actions/runs/6126425700/job/16630411063?pr=2553#step:7:9)535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-focal-cuda12.1-cudnn8-py3-gcc9:4d057f1ef82e7cb2c4dbd891d38b6b63173271af not found: manifest unknown: Requested image not found
Error response from daemon: manifest for 308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-focal-cuda12.1-cudnn8-py3-gcc9:4d057f1ef82e7cb2c4dbd891d38b6b63173271af not found: manifest unknown: Requested image not found
Error response from daemon: manifest for 308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-focal-cuda12.1-cudnn8-py3-gcc9:4d057f1ef82e7cb2c4dbd891d38b6b63173271af not found: manifest unknown: Requested image not found
Error: Process completed with exit code 1.
```

**Review by williamwen42 — APPROVED** [review]

File: recipes_source/torch_compile_backend_ipex.rst [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+=====================================
+
+To work better with `torch.compile`, Intel® Extension for PyTorch* implements a backend `ipex`. It targets to improve hardware resource usage efficiency on Intel platforms for better performance. The `ipex` backend is implemented with further customizations designed in Intel® Extension for PyTorch* for the model compilation.
+
+Usage Example
+~~~~~~~~~~~~~
+
+Train FP32
+----------
+
```
Comment (svekars) [inline recipes_source/torch_compile_backend_ipex.rst:11]: Can we add something like: "The following example demonstrates...." and describe what it demonstrates.

File: recipes_source/torch_compile_backend_ipex.rst [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+       optimizer.zero_grad()
+       output = compile_model(data)
+       loss = criterion(output, target)
+       loss.backward()
+       optimizer.step()
+
+
+Train BF16
+----------
+
```
Comment (svekars) [inline recipes_source/torch_compile_backend_ipex.rst:61]: Same as above - can we add something like: "The following example demonstrates...." and explain what it demonstrates

File: recipes_source/torch_compile_backend_ipex.rst [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+           optimizer.zero_grad()
+           output = compile_model(data)
+           loss = criterion(output, target)
+           loss.backward()
+           optimizer.step()
+
+
+Inference FP32
+--------------
+
```
Comment (svekars) [inline recipes_source/torch_compile_backend_ipex.rst:112]: Same as above - can we add something like: "The following example demonstrates...." and explain what it demonstrates

File: recipes_source/torch_compile_backend_ipex.rst [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+   compile_model = torch.compile(model, backend="ipex")
+   ######################################################
+
+   with torch.no_grad():
+       compile_model(data)
+
+
+Inference BF16
+--------------
+
```
Comment (svekars) [inline recipes_source/torch_compile_backend_ipex.rst:137]: Same as above - can we add something like: "The following example demonstrates...." and explain what it demonstrates

File: recipes_source/torch_compile_backend_ipex.rst [hunk 5]
```diff
@@ -0,0 +1,157 @@
+Intel® Extension for PyTorch* Backend
+=====================================
+
+To work better with `torch.compile`, Intel® Extension for PyTorch* implements a backend `ipex`. It targets to improve hardware resource usage efficiency on Intel platforms for better performance. The `ipex` backend is implemented with further customizations designed in Intel® Extension for PyTorch* for the model compilation.
```
Comment (svekars) [inline recipes_source/torch_compile_backend_ipex.rst:4]:
```suggestion
To work better with `torch.compile`, Intel® Extension for PyTorch* implements a backend ``ipex``. 
It targets to improve hardware resource usage efficiency on Intel platforms for better performance.
The `ipex` backend is implemented with further customizations designed in Intel® Extension for
PyTorch* for the model compilation.
```

**Review by svekars — COMMENTED** [review]
Looks good, added a couple of minor editorial requests.

File: recipes_source/torch_compile_backend_ipex.rst (hunk 1 above)
Comment (jingxu10) [reply recipes_source/torch_compile_backend_ipex.rst:14]: Sure. Added description below.

File: recipes_source/torch_compile_backend_ipex.rst (hunk 2 above)
Comment (jingxu10) [reply recipes_source/torch_compile_backend_ipex.rst:66]: Sure. Added description below.

File: recipes_source/torch_compile_backend_ipex.rst (hunk 3 above)
Comment (jingxu10) [reply recipes_source/torch_compile_backend_ipex.rst:119]: Sure. Added description below.

File: recipes_source/torch_compile_backend_ipex.rst (hunk 4 above)
Comment (jingxu10) [reply recipes_source/torch_compile_backend_ipex.rst:146]: Sure. Added description below.

**Comment by jingxu10** [comment]
Hi @svekars Would you help to check if the code changes look good?

**Comment by jingxu10** [comment]
Hi @svekars How do the amendments look? Would you help to merge the PR is it looks good? Thank you.

**Review by svekars — APPROVED** [review]
